class-gf-razorpay.php

```diff
@@ -427,4 +451,50 @@ public function billing_info_fields()
 
         return $fields;
     }
+
+    public function redirect_url( $feed, $submission_data, $form, $entry )
+    {
+        //updating lead's payment_status to Processing
+        GFAPI::update_entry_property( $entry['id'], 'payment_status', 'Processing' );
+
+        $this->generate_razorpay_order($entry, $form);
+    }
+
+    public function init()
+    {
+        add_filter( 'gform_notification_events', array( $this, 'notification_events' ), 10, 2 );
+
+        add_filter( 'gform_post_payment_action', array( $this, 'post_payment_action' ), 10, 2 );
+
+        // Supports frontend feeds.
+        $this->_supports_frontend_feeds = true;
+
+        parent::init();
+
+    }
+
+    // Added custom event to provide option to chose event to send notifications.
+    public function notification_events($notification_events, $form)
+    {
+        $has_razorpay_feed = function_exists( 'gf_razorpay' ) ? gf_razorpay()->get_feeds( $form['id'] ) : false;
+
+        if ($has_razorpay_feed) {
+            $payment_events = array(
+                'complete_payment'          => __('Payment Completed', 'gravityforms'),
```

Comment:
Currently notification works only for one event at a time. So we should provide option for multiple event notifications or just provide only one default option event.